components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -288,14 +288,19 @@ void ConversationHandler::BindUntrustedConversationUI(
 }
 
 void ConversationHandler::OnConversationMetadataUpdated() {
-  // Pass the updated data to archive content
   if (archive_content_) {
-    archive_content_->SetMetadata(
-        metadata_->associated_content->url.value_or(GURL()),
-        base::UTF8ToUTF16(metadata_->associated_content->title.value_or("")),
-        metadata_->associated_content->content_type ==
-            mojom::ContentType::VideoTranscript);
+    if (metadata_->associated_content) {
+      // Pass the updated data to archive content
+      archive_content_->SetMetadata(
+          metadata_->associated_content->url,
+          base::UTF8ToUTF16(metadata_->associated_content->title),
+          metadata_->associated_content->content_type ==
+              mojom::ContentType::VideoTranscript);
+    } else {
+      archive_content_->SetMetadata(GURL(), u"", false);
+    }
```

Comment:
I still think maybe it's better to reset `archive_content_` (and `associated_content_delegate_`) here, otherwise this reads as confusing.